language/move-prover/boogie-backend/src/bytecode_translator.rs

```diff
@@ -1260,6 +1262,26 @@ impl<'env> FunctionTranslator<'env> {
                             str_local(src)
                         );
                     }
+                    CastU16 => {
+                        let src = srcs[0];
+                        let dest = dests[0];
+                        emitln!(
+                            writer,
+                            "call {} := $CastU16({});",
+                            str_local(dest),
+                            str_local(src)
+                        );
+                    }
+                    CastU32 => {
+                        let src = srcs[0];
+                        let dest = dests[0];
+                        emitln!(
+                            writer,
+                            "call {} := $CastU32({});",
+                            str_local(dest),
+                            str_local(src)
+                        );
+                    }
```

Comment:
Looks correct to me. cc @wrwg @rahxephon89 @junkil-park  if you wish to double check

Reply:
We version-gate this. Doesn't that work?

Reply:
I'm sure @rahxephon89 already checked this. See
https://github.com/move-language/move/pull/547#discussion_r998915512